### PR TITLE
Feature/alcs 494 - part 2

### PR DESCRIPTION
**Pull request by mhuseinov**
- basic decision fields
- support for nfu and preparation for the next types
- CRUD
- Validation
- unit tests

### CODE REVIEW

File: alcs-frontend/src/app/features/application/decision/decision-v2/decision-input/decision-input-v2.component.ts [hunk 1]
```diff
@@ -80,6 +86,7 @@ export class DecisionInputV2Component implements OnInit, OnDestroy {
     daysHideFromPublic: new FormControl<number>(2, [Validators.required]),
     rescindedDate: new FormControl<Date | undefined>(undefined),
     rescindedComment: new FormControl<string | undefined>(undefined),
+    // components: this.fb.array([]),
```
Comment (dhaselhan) [inline alcs-frontend/src/app/features/application/decision/decision-v2/decision-input/decision-input-v2.component.ts:89]: Left over?

File: alcs-frontend/src/app/features/application/decision/decision-v2/decision-input/decision-input-v2.component.ts [hunk 2]
```diff
@@ -90,7 +97,8 @@ export class DecisionInputV2Component implements OnInit, OnDestroy {
     private route: ActivatedRoute,
     private toastService: ToastService,
     private applicationSubmissionService: ApplicationSubmissionService,
-    public dialog: MatDialog
+    public dialog: MatDialog,
+    private fb: FormBuilder
```
Comment (dhaselhan) [inline alcs-frontend/src/app/features/application/decision/decision-v2/decision-input/decision-input-v2.component.ts:101]: Left over formBuilder?

**Review by dhaselhan — APPROVED** [review]
LGTM 👍

File: alcs-frontend/src/app/features/application/decision/decision-v2/decision-input/decision-input-v2.component.ts (hunk 2 above)
Comment (mhuseinov) [reply alcs-frontend/src/app/features/application/decision/decision-v2/decision-input/decision-input-v2.component.ts:101]: yea, I'll remove it, thanks!

File: alcs-frontend/src/app/features/application/decision/decision-v2/decision-input/decision-input-v2.component.ts (hunk 1 above)
Comment (mhuseinov) [reply alcs-frontend/src/app/features/application/decision/decision-v2/decision-input/decision-input-v2.component.ts]: yep, removed

**Pull request merged**